### ISSUE TITLE
Investigate python pickle logic failures on Hera and WCOSS

### ISSUE THREAD

**Issue by jprestop** [dtcenter/MET#1264]
Straight python embedding works but python pickle logic does not work on Hera and WCOSS.

Example on WCOSS
_________________

Load modules and set env var:

module use /gpfs/dell2/emc/verification/noscrub/Julie.Prestopnik/modulefiles
module load met/9.0_beta3
export MET_PYTHON_EXE=/usrx/local/prod/packages/python/3.6.3/bin/python

Run:

ascii2nc "/gpfs/dell2/emc/verification/noscrub/Julie.Prestopnik/met/9.0_beta3/share/met/python/read_ascii_point.py /gpfs/dell2/emc/verification/noscrub/Julie.Prestopnik/met/9.0_beta3/met-9.0_beta3/data/sample_obs/ascii/sample_ascii_obs.txt" ascii2nc_pickle.nc -format python

Output:

DEBUG 1: Config File: /gpfs/dell2/emc/verification/noscrub/Julie.Prestopnik/met/9.0_beta3/share/met/config/Ascii2NcConfig_default
Python Script:     /gpfs/dell2/emc/verification/noscrub/Julie.Prestopnik/met/9.0_beta3/share/met/wrappers/point_write_pickle.py
User Command:     ['/gpfs/dell2/emc/verification/noscrub/Julie.Prestopnik/met/9.0_beta3/share/met/python/read_ascii_point.py', '/gpfs/dell2/emc/verification/noscrub/Julie.Prestopnik/met/9.0_beta3/met-9.0_beta3/data/sample_obs/ascii/sample_ascii_obs.txt']
Write Pickle:     /tmp/tmp_ascii2nc_pickle_169858_0
Python Script:     /gpfs/dell2/emc/verification/noscrub/Julie.Prestopnik/met/9.0_beta3/share/met/python/read_ascii_point.py
Input File:    '/gpfs/dell2/emc/verification/noscrub/Julie.Prestopnik/met/9.0_beta3/met-9.0_beta3/data/sample_obs/ascii/sample_ascii_obs.txt'
Data Length:    2140
Data Type:    <class 'list'>
ERROR  :
ERROR  : Python3_Script::Python3_Script(const char *) -> unable to open script "generic_pickle"
ERROR  :
Exception ignored in: <module 'threading' from '/usrx/local/prod/packages/python/3.6.3/lib/python3.6/threading.py'>
Traceback (most recent call last):
  File "/usrx/local/prod/packages/python/3.6.3/lib/python3.6/threading.py", line 1289, in _shutdown
    assert tlock.locked()
SystemError: <built-in method locked of _thread.lock object at 0x2b2240ea7dc8> returned a result with an error set

**Comment by jprestop** [comment]
Adding PyErr_Print(); to src/libcode/vx_python3_utils/python3_script.cc above mlog in the following code:

> if ( ! Module )  {
>   mlog << Error << "\nPython3_Script::Python3_Script(const char *) -> "
>        << "unable to open script \"" << path << "\"\n\n";
>   Py_Finalize();
>   exit ( 1 );
>}

and then recompiling and rerunning produced the following information:

> AttributeError: module 'sys' has no attribute 'argv'

See:
https://bugs.python.org/issue32573

Adding the following code to data/wrappers/generic_pickle.py resolved the issue:

>  Needed to avoid the error:
>  AttributeError: module 'sys' has no attribute 'argv'
>  Since `argv` contains the list of **command line** arguments passed to the
>  Python interpreter, `argv` is not available within embedded interpreters, and 
>  scripts that run in embedded environments will raise an `AttributeError` when
>  they attempt to access `argv`. As a workaround, assign a list with an empty 
>  string manually: 
> if not hasattr(sys, 'argv'):
>    sys.argv  = ['']

**Comment by JohnHalleyGotway** [comment]
We chose to fix this issue by splitting the generic_pickle.py script out into 2 scripts: generic_pickle.py and read_pickle_dataplane.py

The new generic_pickle.py script will never be run with arguments, while the new read_pickle_dataplane.py script is always run with arguments.  With that change, we'll avoid this python bug.

At the same time, we modified the python wrapper scripts to make their naming conventions more consistent and include a standard header at the top.

Committed changes to the develop branch and ran the unit_python.xml tests to ensure they work.

**Comment by JohnHalleyGotway** [comment]
In testing, I also found that employing the pickle logic to run a python script from the current working directory fails.  The problem is in the write_pickle*.py scripts.  They call os.path.dirname() which returns an empty string instead of "." when the script is in the current directory.  Updated those scripts using:
   pyembed_module_dir  = os.path.dirname(sys.argv[2]) or '.'

This module importing logic in these scripts could also be replaced by the importlib module:
https://github.com/NCAR/MET/issues/1265
